Gems/DebugDraw/Code/Source/DebugDrawSystemComponent.h

```diff
@@ -141,15 +144,35 @@ namespace DebugDraw
         template <typename F>
         void removeExpiredDebugElementsFromVector(AZStd::vector<F>& vectorToExpire);
 
+        struct DebugDrawSphereElementWrapper : DebugDrawSphereElement
+        {
+            AZ::Vector3    m_previousWorldLocation = AZ::Vector3::CreateZero();
+            float          m_previousRadius = 0;
+            uint32_t       m_localInstanceIndex;
+        };
+
+        struct DebugDrawObbElementWrapper : DebugDrawObbElement
+        {
+            AZ::Vector3    m_previousWorldLocation;
+            AZ::Vector3    m_previousScale;
+            AZ::Quaternion m_previousRotation;
+        };
+
+        void CheckAddRaytracingData(DebugDrawSphereElementWrapper& element);
+        void CheckAddRaytracingData(DebugDrawObbElementWrapper& element);
+
+        void CheckRemoveRaytracingData(const DebugDrawSphereElementWrapper& element);
+        void CheckRemoveRaytracingData(const DebugDrawObbElementWrapper& element);
```

Comment:
Why are they called CheckAdd* and CheckRemove*? They are not even returning a boolean to indicate if the check succeeded or failed. Why not simply use AddRaytracingData and RemoveRaytracingData? Lets also add a comment explaining the purpose of these methods here.

Reply:
Renamed to AddRayTracingData/RemoveRayTracingData